ff-fft/src/polynomial/dense.rs

```diff
@@ -143,10 +158,22 @@ impl<F: PrimeField> DensePolynomial<F> {
     pub fn mul_by_vanishing_poly(&self, domain: EvaluationDomain<F>) -> DensePolynomial<F> {
         let mut shifted = vec![F::zero(); domain.size()];
         shifted.extend_from_slice(&self.coeffs);
-        shifted
```

Comment:
Let's do something similar to here: https://github.com/scipr-lab/zexe/blob/8a9343ad03e24cf278598e19c9a7b928b3ab729e/algebra/src/msm/fixed_base.rs#L80

This avoid duplicating the rest of the iterator chain.

Reply:
fixed